### DISCUSSION TITLE
SAX parsing....

### DISCUSSION THREAD

**Discussion by Alexandring** [simdjson/simdjson#1926] [Q&A] [answered]
Hi! I'm trying to understand how the library works and I can't find an example with SAX parsing. I have a byte stream, which is a json document. The data is coming in gradually. Could you please indicate how to use the library in this case?
How do I get the value in the parsing process? Do I need to define my stream class and callback functions?

**Comment by lemire** [comment, marked as answer]
_The data is coming in gradually._ 

The simdjson library is for in-memory parsing. We expect the JSON to reside in a string which is in memory. You can, of course, use memory-file mapping as well.

So just load all of your data into memory and then use the simdjson library.

Regarding the API, we don't have a SAX model (though @jkeiser did craft one at one point): instead of we recommend On Demand which we feel is a superior solution. You still need to have the JSON document be a string.

If you have a stream of JSON documents, then we have an API for that as well, but it is expected that the data is in memory as well.
